chanfitness/chaneventstore.go

```diff
@@ -352,65 +301,72 @@ func (c *ChannelEventStore) consume(subscriptions *subscriptions) {
 	}
 }
 
-// GetLifespan returns the opening and closing time observed for a channel and
-// a boolean to indicate whether the channel is known the the event store. If
-// the channel is still open, a zero close time is returned.
-func (c *ChannelEventStore) GetLifespan(
-	channelPoint wire.OutPoint) (time.Time, time.Time, error) {
+// ChannelInfo provides the set of information that the event store has recorded
+// for a channel.
+type ChannelInfo struct {
+	// Lifetime is the total amount of time we have monitored the channel
+	// for.
+	Lifetime time.Duration
+
+	// Uptime is the total amount of time that the channel peer has been
+	// observed as online during the monitored lifespan.
+	Uptime time.Duration
```

Comment:
Same question in a different form, but if a peer reportedly comes online such that it's always in the idle timeout then what is the uptime?

Reply:
So this would be the case where a peer connects to us, does not respond to any messages for the idle timeout, we disconnect and then they immediately reconnect but don't respond to any messages again?

With the current notifications, this peer would be up for that period of time, since we've defined online as having a connection open with that peer (so far). I think this case would likely be due to a mis-implemented peer, and while somebody could setup their node to "game" our uptime metric, the flap count would reflect that they go down every 5 minutes, and I don't think there's much to gain because they would have to respond to our messages to participate in forwards (the only incentive I can see for wanting high uptime, we choose to route through you).  

We could look at moving notifications for uptime tracking to a lower level, but I think we will then start running into ordering issues.